src/server/pkg/pool/pool.go

```diff
@@ -81,15 +83,13 @@ func (p *Pool) updateAddresses(endpoints *api.Endpoints) {
 		// According the k8s docs, the full set of endpoints is the cross
 		// product of (addresses x ports).
 		for _, address := range subset.Addresses {
-			for _, port := range subset.Ports {
-				addr := fmt.Sprintf("%s:%d", address.IP, port.Port)
-				if cc := p.conns[addr]; cc != nil {
-					addresses[addr] = cc
-				} else {
-					// we don't actually connect here because there's no way to
-					// return the error
-					addresses[addr] = &connCount{}
-				}
```

Comment:
I changed this because it was causing a really confusing bug where the master sometimes couldn't connect to the workers because I added an extra port to the service for exposing the users service. I wound up making a separate k8s service for it so I could do nodeports but this was a confusing enough bug that I wanted to change it so someone else doesn't lose time to it.